### PR TITLE
Fix Mllama vision/cross-attention being treated as causal under SDPA

### LINKED ISSUE #48172: [Mllama] MllamaVisionAttention and MllamaTextCrossAttention never set self.is_causal — cross-attention is silently causal under sdpa
### System Info

- transformers @ `0c92811846` (main, editable), also present in 5.15.x
- torch 2.13.0, Python 3.13.14, macOS 26.4 arm64 (CPU — no GPU needed for the repro below)

### Who can help?

@zucchini-nlp @molbap

### Reproduction

`MllamaVisionAttention` and `MllamaTextCrossAttention` never set `self.is_causal`, but `MllamaPreTrainedModel` declares `_supports_sdpa = True` / `_supports_flash_attn = True` (`modeling_mllama.py:773`). Only `MllamaTextSelfAttention` sets it (`:502`).

Because `sdpa_attention.py:108` resolves the flag as `getattr(module, "is_causal", True)`, a module that never declares it is treated as a decoder. The result is that **cross-attention gets a causal mask whenever `cross_attention_mask` is not supplied.**

Instrumenting `F.scaled_dot_product_attention` to log the flag that actually reaches the kernel, on `hf-tiny-v2/tiny-random-MllamaForConditionalGeneration` with `attn_implementation="sdpa"`:

```
static check:
  MllamaVisionAttention        is_causal = **MISSING**
  MllamaTextSelfAttention      is_causal = True
  MllamaTextCrossAttention     is_causal = **MISSING**

forward with cross_attention_mask=None:
  MllamaTextCrossAttention     is_causal=True   mask_None=True    <-- causal mask on cross-attention
  MllamaTextSelfAttention      is_causal=True   mask_None=True
  MllamaVisionAttention        is_causal=False  mask_None=False

forward with cross_attention_mask supplied:
  MllamaTextCrossAttention     is_causal=False  mask_None=False
  MllamaVisionAttention        is_causal=False  mask_None=False
```

The vision tower is unaffected in practice only because `_prepare_aspect_ratio_attention_mask` (`:951`) always materialises a 4-D mask, so `sdpa_attention.py:124` force-clears the flag. The bug is present there too, just currently masked.

For cross-attention there is no such net: `:1354` is `if cross_attention_mask is not None:` with an `else`, so `None` reaches `MllamaTextCrossAttention` at `:688` and the flag defaults to `True`.

I have **not** reproduced the FlashAttention path (no CUDA here). By inspection, `flash_attention.py:74` reads `module.is_causal` as a bare attribute rather than via `getattr`, so FA2 should raise `AttributeError` on these two modules — same as #39231 did for Qwen2.5-VL.

### Expected behavior

Cross-attention is never causal, and neither is a vision tower. Both should declare `self.is_causal = False`, matching siglip (`:266`), CLIP (`:291`) and Qwen2-VL (`:354`).

This is the same defect that #39231 reported for Qwen2-VL / Qwen2.5-VL, fixed in `7a25f8dfdb` (#39121) by adding exactly `self.is_causal = False` to the vision attention, and shipped in v4.53.1. Mllama looks to have been missed.

```diff
  class MllamaVisionAttention(nn.Module):
      def __init__(self, config):
+         self.is_causal = False

  class MllamaTextCrossAttention(nn.Module):
      def __init__(self, config, layer_idx):
+         self.is_causal = False
```

Happy to open a PR if this looks right.

### PR DESCRIPTION
**Pull request by tanvir-ux**
[CPU CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48218) [GPU run-slow](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48218)

# What does this PR do?

Fixes #48172.

sdpa_attention resolves causality as getattr(module, "is_causal", True). MllamaTextSelfAttention sets is_causal = True, but MllamaVisionAttention and MllamaTextCrossAttention never set the flag. With no 4-D mask (typical for text cross-attention when cross_attention_mask is omitted), SDPA therefore applies a causal mask to encoder/cross-attention.

This is the same class of miss as Qwen2.5-VL (#39231 / #39121). The FlashAttention path reads module.is_causal as a required attribute, so those two modules can also raise AttributeError under FA2.

Change: set self.is_causal = False on both modules, matching CLIP / SigLIP / Qwen2-VL vision attention. Added a unit test that instantiates the three attention classes and checks the flag.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the contributor guideline and the Pull Request checks?
- [x] Was this discussed/approved via a Github issue or the forum? https://github.com/huggingface/transformers/issues/48172
- [ ] Did you make sure to update the documentation with your changes according to the guidelines?
- [x] Did you write any new necessary tests?

## Who can review?

@zucchini-nlp

### CODE REVIEW

**Comment by zucchini-nlp** [comment]
See comment under the issue
